src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -489,13 +494,25 @@ private Field createBatchHydrationTopLevelField(ExecutionContext executionContex
         ListExecutionResultNode listResultNode = (ListExecutionResultNode) root;
         List<ExecutionResultNode> resolvedNodes = listResultNode.getChildren();
 
+        if (isResolveByIndex) {
+            assertTrue(resolvedNodes.size() == hydrationInputNodes.size(), () -> "Expect the size of the resolved nodes to match the size of the input when resolving by index");
```

Comment:
Can we make this assertion more human informative

`"If you use indexed hydration then you MUST follow a contract where the resolved nodes matches the size of the input arguments.  We expected %d returned nodes but only got %d"

Reply:
Sure :)

- [ ] Add node sizes to error message